reconcile/gitlab_housekeeping.py

```diff
@@ -202,19 +203,41 @@ def is_rebased(mr, gl: GitLabApi) -> bool:
 
 
 def get_labels(mr: ProjectMergeRequest, gl: GitLabApi) -> list[str]:
-    labels = mr.attributes.get("labels")
-    if not labels:
-        # Sometimes the label attribute is empty but shouldn't. Try it again by fetching this MR separately
```

Comment:
So far as I can remember, I've seen that sometimes the `label` attribute is empty, and I tried to mitigate that by fetching it separately, for sure at higher costs :( I don't remember the exact MR or the GitLab version anymore. Let's introduce this change, including your caching approach, and will we see.

Reply:
I could also add a second check for `labels` being truthy and if it is, then log a warning message. If I'm right, this would happen very little if ever. And if I'm wrong, then we'll just have a bunch of unnecessary log messages which we can revert easily.

Reply:
#3053 will handle adding the log message to get more data before making this change.

Reply:
Sounds like a good plan!

Reply:
The outcome of this discussion was ultimately #3066. In the end, it seems like this isn't an issue any longer after observing 7 days of data.